### ISSUE TITLE
HTTP status code for REST Web Services

### ISSUE THREAD

**Issue by niccord** [RotherOSS/otobo#956]
Hi folks!
I am trying to implement some REST web service for otobo, but I notice I can only return 200 or 500 as status codes.

After digging into the code a little, I found some case where give different status code is possible, but not handled.

It would be really nice to allow us to specify the HTTP status code, with something like:

``` perl
return {
    Success => 0,
    HTTPStatus => 401,
    ErrorMessage => 'User can\'t access this resource',
};

return {
    Success => 1,
    HTTPStatus => 201,
    Data    => {
        ID => $ID,
    }
};
```

(I already have some working code in my pocket 🤓)

**Comment by svenoe** [comment]
Hey Nicola,

the generic interface could need a whole set of added functionality. Some time ago I opened an issue for this: https://github.com/RotherOSS/otobo/issues/772. (The last months have been extremely busy, so much not totally critical stuff got delayed.) At the moment I'm thinking all this stuff should be gathered first to have the possibility to unify it a bit, if necessary, and then be released as part of 10.1.

(As always, we are happy about pull requests - this time though, it would be nice, if you could create it for the branch https://github.com/RotherOSS/otobo/tree/issue-%23772-enhance_generic_interface - there I can be more free to just accept stuff right away, if it generally looks good, and then include it together with the rest at a later point in time. :)

Best regards, Sven

**Comment by svenoe** [comment]
I merged the pr - thanks. :)

(A slightly longer reply here: https://github.com/RotherOSS/otobo/pull/968#issuecomment-827371196 )
